pennylane/_qubit_device.py

```diff
@@ -477,24 +477,25 @@ def marginal_prob(self, prob, wires=None):
             return prob
 
         wires = Wires(wires)
-        # get indices of wires on the device's register
-        wire_indices = self.register.indices(wires)
-
         # determine which subsystems are to be summed over
-        inactive_registers = list(set(range(self.num_wires)) - set(wire_indices))
+        inactive_wires = Wires.unique_wires([self.wires, wires])
+
+        # translate to wire labels used by device
+        wires = self.map_wires(wires)
+        inactive_wires = self.map_wires(inactive_wires)
 
         # reshape the probability so that each axis corresponds to a wire
         prob = self._reshape(prob, [2] * self.num_wires)
 
         # sum over all inactive wires
-        prob = self._flatten(self._reduce_sum(prob, inactive_registers))
+        prob = self._flatten(self._reduce_sum(prob, inactive_wires.labels))
 
         # The wires provided might not be in consecutive order (i.e., wires might be [2, 0]).
         # If this is the case, we must permute the marginalized probability so that
         # it corresponds to the orders of the wires passed.
-        basis_states = np.array(list(itertools.product([0, 1], repeat=len(wire_indices))))
+        basis_states = np.array(list(itertools.product([0, 1], repeat=len(wires))))
         perm = np.ravel_multi_index(
-            basis_states[:, np.argsort(np.argsort(wire_indices))].T, [2] * len(wire_indices)
+            basis_states[:, np.argsort(np.argsort(wires.toarray()))].T, [2] * len(wires)
```

Comment:
(following on from my previous comment, if `__array__` is defined, then you will be able to simply call `np.argsort(wires)`)

Reply:
Nice!